aws/sagemaker/llm-instance-recommend/lmi_recommender/lmi_config_recommender.py

```diff
@@ -32,6 +32,21 @@
     }
 }
 
+dynamic_batch_deepspeed = {
+    'smoothquant': {'model_category': {'llama-13b', 'llama-7b', 'gptneox', 'bloom', 'gptj'},
+               'instance': ['g4', 'g5', 'p4', 'p5']},
+    'strategy': {0.1: {"batch_size": 4},
+                 0.3: {"batch_size": 8},
+                 0.5: {"batch_size": 32},
+                 0.7: {"batch_size": 64}},
+    'int8': {
+        "engine": "DeepSpeed",
+        "option.dtype": "fp16",
```

Comment:
Is it important to allow for specifying `bf16` along with smoothquant here?

Reply:
This is not something we should just choose by default. The customer should choose.

Whether bf16 works better than fp16 depends on the model.

My point is, choosing "quantized" vs "not quantized" is an independent choice from "dtype fp16" or "dtype bf16".

Traditional DeepSpeed API used "dtype int8" to mean "use quantization", but in their actual code what they really did was set "dtype" to fp16 and use int8-quantized linear layers. Hijacking dtype is really the wrong API for this.

Reply:
This is probably fine for now, we could always make it more flexible later I suppose.